app/database/item_store.go

```diff
@@ -69,9 +64,32 @@ func (s *ItemStore) HasManagerAccess(user *User, itemIDs ...int64) (hasAccess bo
 	for _, itemID := range itemIDs {
 		idsMap[itemID] = true
 	}
-	err = s.GroupItems().MatchingUserAncestors(user).
+	err = s.PermissionsGenerated().MatchingUserAncestors(user).
 		WithWriteLock().
-		Where("item_id IN (?) AND (cached_manager_access OR owner_access)", itemIDs).
+		Where("item_id IN (?)", itemIDs).
+		Where("can_grant_view_generated_value >= ?", s.PermissionsGranted().GrantViewIndexByKind("content")).
+		Select("COUNT(DISTINCT item_id)").Count(&count).Error()
```

Comment:
Yes there is a test in TestItemStore_CanGrantViewContentOnAll(): ```		{name: "two items (not unique)", ids: []int64{12, 13, 12, 13}, userID: 110, wantResult: true},```
Since the result is true for the test, DISTINCT works fine. The query is ```SELECT COUNT(DISTINCT item_id) FROM `permissions_generated` JOIN (SELECT * FROM `groups_ancestors`  WHERE (`groups_ancestors`.child_group_id = 110) AND (NOW() < `groups_ancestors`.expires_at)) AS ancestors ON permissions_generated.group_id = ancestors.ancestor_group_id WHERE (item_id IN (12,13,12,13)) AND (can_grant_view_generated_value >= 2) FOR UPDATE```